internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,159 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource struct {
```

Comment:
I think I'd recommend exporting this type, at least if we're going to return this type from the `New` function. The other option is to return `input.EntitySource`.

Returning an unexported type can be awkward for callers because they can use it in the scope of where it was returned but they can't pass it to other functions because they can't refer to it due to it being unexported.

Reply:
I feel like I might have experienced the awkwardness myself somewhere, but I am not sure if it was exactly this? With what I have in my mind right at this moment, this is just the OOPs concept of "Objects should not be directly accessed by anything outside the package it's defined in, but should be accessible to the world only through exported methods, that can control what information is exposed to the world"

Reply:
But having said that, I don't think I've heard that principle repeated in the context of Golang yet though. I'm still inclined to keep it unexported, and export it when we see we need to (and that'll be added to my TIL moments too), but if you insist on doing this change now I'm happy to go ahead and do it anyway.

Reply:
We're in `internal` here, so it really doesn't matter all that much since we're the only clients. Let's just go with this for now.